charts/argo-workflows/templates/controller/workflow-controller-deployment.yaml

```diff
@@ -77,12 +77,16 @@ spec:
           ports:
             - name: metrics
               containerPort: {{ .Values.controller.metricsConfig.port }}
+            - containerPort: 6060
           livenessProbe:
```

Comment:
I think this should be configurable. In the current configuration the controller may be down up to four minutes until Kubernetes will detect that, and many people will want a faster reaction. The entire section may be put into the config, with the value moved into in `values.yaml`:
```
  livenessProbe: {{ .Values.controller.livenessProbe | toYaml | nindent 12 }}
```

Reply:
good call. this has been updated